bundles/org.openhab.binding.meteoalerte/src/main/java/org/openhab/binding/meteoalerte/internal/MeteoAlertIconProvider.java

```diff
@@ -0,0 +1,128 @@
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Locale;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.meteoalerte.internal.json.ResponseFieldDTO.AlertLevel;
+import org.openhab.core.i18n.TranslationProvider;
+import org.openhab.core.ui.icon.IconProvider;
+import org.openhab.core.ui.icon.IconSet;
+import org.openhab.core.ui.icon.IconSet.Format;
+import org.osgi.framework.BundleContext;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link MeteoAlertIconProvider} is the class providing binding related icons.
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+@Component(service = { IconProvider.class, MeteoAlertIconProvider.class })
+@NonNullByDefault
+public class MeteoAlertIconProvider implements IconProvider {
+    private static final String DEFAULT_LABEL = "Météo Alerte Icons";
+    private static final String DEFAULT_DESCRIPTION = "Icons illustrating weather events provided by Météo Alerte";
+    private static final Set<String> ICONS = Set.of(WAVE, AVALANCHE, HEAT, FREEZE, FLOOD, SNOW, STORM, RAIN, WIND,
+            "meteo_france");
+
+    private final Logger logger = LoggerFactory.getLogger(MeteoAlertIconProvider.class);
+    private final BundleContext context;
+    private final TranslationProvider i18nProvider;
+
+    @Activate
+    public MeteoAlertIconProvider(final BundleContext context, final @Reference TranslationProvider i18nProvider) {
+        this.context = context;
+        this.i18nProvider = i18nProvider;
+    }
+
+    @Override
+    public Set<IconSet> getIconSets() {
+        return getIconSets(null);
+    }
+
+    @Override
+    public Set<IconSet> getIconSets(@Nullable Locale locale) {
+        String label = getText("label", DEFAULT_LABEL, locale);
+        String description = getText("decription", DEFAULT_DESCRIPTION, locale);
+
+        return Set.of(new IconSet(BINDING_ID, label, description, Set.of(Format.SVG)));
+    }
+
+    private String getText(String entry, String defaultValue, @Nullable Locale locale) {
+        String text = defaultValue;
+        if (locale != null) {
+            text = i18nProvider.getText(context.getBundle(), "iconset." + entry, defaultValue, locale);
+            text = text == null ? defaultValue : text;
+        }
+        return text;
+    }
+
+    @Override
+    public @Nullable Integer hasIcon(String category, String iconSetId, Format format) {
+        return ((ICONS.contains(category) && iconSetId.equals(BINDING_ID))
+                || (ICONS.contains(category.replace(BINDING_ID + ":", "")))) && format == Format.SVG ? 7 : null;
```

Comment:
I was surprised to get an icon using MainUI. Now I understand when I see the code.
I hope we agree that using `category.replace(BINDING_ID + ":", "")` is just a workaround to a bug in Main UI.
It has to be removed as soon as Main UI is fixed.

Reply:
You're correct - this is not intended to stay

Reply:
Keep as it is to not delay the merge but please add a comment on both places mentioning the workaround to the bug in MainUI that do not yet support OH icon set.

Reply:
Ok, done.